core-plugins/src/main/java/io/cdap/plugin/batch/joiner/Joiner.java

```diff
@@ -77,18 +82,22 @@ public Joiner(JoinerConfig conf) {
   public void configurePipeline(MultiInputPipelineConfigurer pipelineConfigurer) {
     MultiInputStageConfigurer stageConfigurer = pipelineConfigurer.getMultiInputStageConfigurer();
     Map<String, Schema> inputSchemas = stageConfigurer.getInputSchemas();
-    init(inputSchemas);
+    // Get failure collector for updated validation API
+    FailureCollector collector = pipelineConfigurer.getMultiInputStageConfigurer().getFailureCollector();
+    init(inputSchemas, collector);
     //validate the input schema and get the output schema for it
-    stageConfigurer.setOutputSchema(getOutputSchema(inputSchemas));
+    stageConfigurer.setOutputSchema(getOutputSchema(inputSchemas, collector));
   }
 
   @Override
   public void prepareRun(BatchJoinerContext context) {
     if (conf.getNumPartitions() != null) {
       context.setNumPartitions(conf.getNumPartitions());
     }
-    init(context.getInputSchemas());
-    Collection<OutputFieldInfo> outputFieldInfos = createOutputFieldInfos(context.getInputSchemas());
+    // Get failure collector for updated validation API
+    FailureCollector collector = context.getFailureCollector();
+    init(context.getInputSchemas(), collector);
+    Collection<OutputFieldInfo> outputFieldInfos = createOutputFieldInfos(context.getInputSchemas(), collector);
```

Comment:
after this line, call collector.getOrThrowException() to stop execution if there are problems.

Reply:
Fixed